Release Topic J RR paper after second fgm (#2125) (#679)
eu-digital-identity-wallet/eudi-doc-architecture-and-reference-framework: docs/discussion-topics/j-rr-wallet-to-wallet-interactions.md

> > *When sending a presentation request to a Holder Wallet Unit, a Verifier Wallet Unit SHOULD include a cryptographic proof of its authenticity and validity.*

Reviewer: ```suggestion
> *When sending a presentation request to a Holder Wallet Unit, a Verifier Wallet Unit SHALL include a cryptographic proof of its authenticity and validity.*
```

> > *Before presenting a received presentation request to its User, a Holder Wallet Unit SHOULD cryptographically verify, within the same protocol session as the presentation request, that the Verifier Wallet Unit is a genuine, non-revoked EUDI Wallet Unit operated by a recognised Wallet Provider.*

Reviewer: Same as above. IMO, this should be a SHALL if the verifier provided such a proof.

> > *Before presenting a received presentation request to its User, a Holder Wallet Unit SHOULD cryptographically verify, within the same protocol session as the presentation request, that the Verifier Wallet Unit is a genuine, non-revoked EUDI Wallet Unit operated by a recognised Wallet Provider.*

Reviewer: ```suggestion
> *Before presenting a received presentation request to its User, a Holder Wallet Unit SHALL cryptographically verify, within the same protocol session as the presentation request, that the Verifier Wallet Unit is a genuine, non-revoked EUDI Wallet Unit operated by a recognised Wallet Provider.*
```

> > *If this verification fails or cannot be performed, the Holder Wallet Unit SHOULD warn the Holder that it could not verify that the Verifier is using an authentic EUDI Wallet Unit, and SHOULD enable the Holder to continue or abort the transaction.*

Reviewer: Same as above. These are two distinct cases:
- if failed -> shall warn
- if not provided / cannot be performed -> should warn but it needs to be described in which situations it is ok when the wallet unit does not warn the user.

Reviewer: essentially, for every SHOULD, there has to be an "unless X", or a description when the SHOULD does not apply, otherwise the requirement is not testable.

Reviewer: ```suggestion
> *If this verification fails or cannot be performed, the Holder Wallet Unit SHALL notify the User. In addition, the Wallet Unit SHALL either not present the requested attributes to the Verifier Wallet Unit, or give the User the choice to present the requested attributes or not. Note: It is up to the Wallet Provider to make a choice for one of these two options.*
```

> All three requirements are phrased as recommendations (SHOULD) rather than mandates (SHALL). Verifying the Verifier Wallet Unit within the same protocol session is broadly seen as desirable, but the benefit may not outweigh the additional complexity, and the appropriate strength of each requirement may also depend on which technical approach is ultimately chosen (see Chapter 4).

Reviewer: ```suggestion
All three requirements are tentatively phrased as mandatory (SHALL).
```

> The wording is deliberately mechanism-agnostic: it fixes the assurance objective (genuine, non-revoked Verifier Wallet Unit, verified within the same protocol session), while leaving the concrete technical realisation to the technical specification. Chapter 4 illustrates four approaches that would meet these requirements and the technical complexity each would involve. The exact phrasing, the HLR indices, and whether any of these SHOULDs should be strengthened to SHALL once a technical approach is selected are open questions for Member State discussion.

Reviewer: ```suggestion
The wording is deliberately mechanism-agnostic: it fixes the assurance objective (genuine, non-revoked Verifier Wallet Unit, verified within the same protocol session), while leaving the concrete technical realisation to the technical specification. Chapter 4 illustrates four approaches that would meet these requirements and the technical complexity each would involve. The exact phrasing, the HLR indices, and whether any of these SHALLs should be weakened to SHOULD once a technical approach is selected are open questions for Member State discussion.
```

| Verifier discloses data before the Holder is authenticated | No (no Verifier data disclosed) | Yes (Verifier PID is sent together with the request) | Yes (Verifier PID is presented in the reversed first run) | No (no Verifier data disclosed) |

Whether any of the approaches should be mandatory, optional, or conditional, and whether a combined model is appropriate, are open questions for Member State discussion.

## 5 Proposed Additions and Changes to HLRs in ARF Annex 2

This chapter turns the candidate phrasings presented in [Section 3.2](#32-proposed-high-level-requirement-for-discussion) into concrete proposed HLRs for ARF Annex 2. The HLRs are formulated at the SHOULD level, consistent with the position developed in [Section 3.2](#32-proposed-high-level-requirement-for-discussion); their strength may be revisited once a technical approach is selected.

### 5.1 New HLRs to be added to Topic 30

The following three HLRs are proposed for addition to Topic 30 (Interaction between Wallet Units) of Annex 2 of ARF v2.8.0. Indexing continues from the current last entry in that version (W2W_22).

| **Index** | **Requirement specification** |
| -- | -- |

> | W2W_23 | When sending a presentation request to a Holder Wallet Unit, a Verifier Wallet Unit SHOULD include in that request a cryptographic proof of its authenticity and validity, bound to the protocol session in which the presentation request is sent. |

Reviewer: We should describe in which situation it is ok that the Verifier does not provide such a cryptographic proof.

> | W2W_23 | When sending a presentation request to a Holder Wallet Unit, a Verifier Wallet Unit SHOULD include in that request a cryptographic proof of its authenticity and validity, bound to the protocol session in which the presentation request is sent. |

Reviewer: ```suggestion
| W2W_23 | When sending a presentation request to a Holder Wallet Unit, a Verifier Wallet Unit SHALL include in that request a cryptographic proof of its authenticity and validity, bound to the protocol session in which the presentation request is sent. |
```

> | W2W_24 | Before presenting a received presentation request to the Holder, a Holder Wallet Unit SHOULD, within the same protocol session as the presentation request, cryptographically verify that the Verifier Wallet Unit is a genuine, non-revoked EUDI Wallet Unit operated by a recognised Wallet Provider. |

Reviewer: I think if the verifier included such a proof, the wallet MUST verify the proof.
```suggestion
| W2W_24 | If the Verifier Wallet Unit included such a cryptographic proof in the presentation request, and before presenting a received presentation request to the Holder, a Holder Wallet Unit SHALL, within the same protocol session as the presentation request, cryptographically verify that the Verifier Wallet Unit is a genuine, non-revoked EUDI Wallet Unit operated by a recognised Wallet Provider. |
```

> | W2W_24 | Before presenting a received presentation request to the Holder, a Holder Wallet Unit SHOULD, within the same protocol session as the presentation request, cryptographically verify that the Verifier Wallet Unit is a genuine, non-revoked EUDI Wallet Unit operated by a recognised Wallet Provider. |

Reviewer: ```suggestion
| W2W_24 | Before presenting a received presentation request to the Holder, a Holder Wallet Unit SHALL, within the same protocol session as the presentation request, cryptographically verify that the Verifier Wallet Unit is a genuine, non-revoked EUDI Wallet Unit operated by a recognised Wallet Provider. |
```

> | W2W_25 | If the verification required by W2W_24 fails or cannot be performed, a Holder Wallet Unit SHOULD warn the Holder that it could not verify the authenticity of the Verifier Wallet Unit, and SHOULD enable the Holder to continue or abort the transaction. |

Reviewer: I think these might be two distinct requirements.
- if W2W_24 fails, a Holder Wallet Unit SHALL warn the Holder ...
- if W2W_24 cannot be performed (because no such proof was included), a Holder Wallet Unit SHOULD warn the Holder ... in this case, we need to describe in which situation it is ok that the Holder Wallet Unit does not warn the Holder.

> | W2W_25 | If the verification required by W2W_24 fails or cannot be performed, a Holder Wallet Unit SHOULD warn the Holder that it could not verify the authenticity of the Verifier Wallet Unit, and SHOULD enable the Holder to continue or abort the transaction. |

Reviewer: ```suggestion
| W2W_25 | If the verification required by W2W_24 fails or cannot be performed, a Holder Wallet Unit SHALL notify the User. In addition, the Wallet Unit SHALL either not present the requested attributes to the Verifier Wallet Unit, or give the User the choice to present the requested attributes or not. Note: It is up to the Wallet Provider to make a choice for one of these two options. |
```
